packages/amesos2/src/Amesos2_TpetraCrsMatrix_MatrixAdapter_def.hpp

```diff
@@ -347,9 +357,14 @@ namespace Amesos2 {
                 int i = perm_l2g(row);
                 for (int k=pointers_t(i); k<pointers_t(i+1); k++) {
                   int col = indices_t(k);
-                  transpose_map(k) = pointers(1+col);
-                  indices(pointers(1+col)) = row;
-                  pointers(1+col) ++;
+                  if (col < nRows) {
+                    transpose_map(k) = pointers(1+col);
+                    indices(pointers(1+col)) = row;
+                    pointers(1+col) ++;
+                  } else {
+                    // extra columns
+                    transpose_map(k) = -1;
```

Comment:
It looks like `transpose_map` is a templated type `host_ordinal_type_array`, is this always a signed type (since assigning -1 to it)?